### PR TITLE
Support MacOS using CMake and brew

### PR DESCRIPTION
**Pull request by robromano**
I like this project, but wanted it to build and run on MacOS. So here are some simple changes and updates to the README to build on MacOS Mojave.

There is a compile issue I had, which required some missing #include <array> updates to fix

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+cmake_minimum_required(VERSION 3.8)
+
+file(GLOB SOURCES "*.cpp")
+file(GLOB PAGES "pages/*.cpp")
+file(GLOB API "api/*.cpp")
+
+add_executable(budget ${SOURCES} ${PAGES} ${API})
+
+target_link_libraries(budget OpenSSL::SSL)
+
```
Comment (wichtounet) [inline src/CMakeLists.txt]: 💄 Could you remove the trailing empty lines?

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+#endif()
+
+#if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+#  set(CMAKE_SHARED_LIBRARY_CREATE_CXX_FLAGS "${CMAKE_SHARED_LIBRARY_CREATE_CXX_FLAGS} -undefined dynamic_lookup")
+#endif()
+
+include_directories(${OPENSSL_INCLUDE_DIR})
+
+add_subdirectory(src)
+
```
Comment (wichtounet) [inline CMakeLists.txt]: 💄 Could you remove the trailing empty lines?

File: README.rst [hunk 3]
```diff
@@ -74,11 +73,39 @@ You just have to use make to build it::
     $ make
     $ sudo make install
 
-Man pages and ZSH/Bash completions are also available.
+MacOS
+=====
+
+Using CMake and brew, build from sources.  First, use brew to install required dependencies.  Assumes
```
Comment (wichtounet) [inline README.rst]: Could you add a comment that MacOS support is experimental ?

File: src/pages/expenses_pages.cpp [hunk 4]
```diff
@@ -189,7 +191,7 @@ void budget::time_graph_expenses_page(const httplib::Request& req, httplib::Resp
     ss << "{ name: '12 months average',";
     ss << "data: [";
 
-    std::array<budget::money, 12> average_12;
+	std::array<budget::money, 12> average_12;
```
Comment (wichtounet) [inline src/pages/expenses_pages.cpp]: Could you fix the alignment to four spaces?

File: CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+include_directories(cpp-httplib)
+
+find_package(OpenSSL REQUIRED)
+#if(OPENSSL_FOUND)
+#set(OPENSSL_USE_STATIC_LIBS TRUE)
+#endif()
+
+#if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+#  set(CMAKE_SHARED_LIBRARY_CREATE_CXX_FLAGS "${CMAKE_SHARED_LIBRARY_CREATE_CXX_FLAGS} -undefined dynamic_lookup")
+#endif()
```
Comment (wichtounet) [inline CMakeLists.txt]: If they are not useful, could you remove the commend lines?

File: README.rst [hunk 6]
```diff
@@ -74,11 +73,39 @@ You just have to use make to build it::
...
+    $ ./build_macos.sh
+    $ make
+
+	
+Windows
+=======
+
+If there is some interest on using it on Windows, it
+should not take too much work to port it to Windows. Just le met me know (or
+make a Pull Request with the changes ;) ). .
```
Comment (wichtounet) [inline README.rst]: Good comment 👍

**Comment by robromano** [comment]
I made the fixes, and a few other things. I was trying to change the server domain and didn't understand the difference between ~/.budgetrc and ~/budget/config.  (I learned that the server_listen directive needs to be in ~/.budgetrc.)  In the process of figuring that out I made a few enhancements in the config API.

File: src/config.cpp [hunk 7]
```diff
@@ -37,6 +37,7 @@ bool load_configuration(const std::string& path, config_type& configuration){
 
         if (file.is_open() && file.good()) {
             std::string line;
+            std::cout << "INFO: Reading config file " << path << std::endl;
```
Comment (wichtounet) [inline src/config.cpp]: Unfortunately, we have to remove this output line. If we use the tool in command line mode, this line will appear every single time. I only output stuff when there is an error or when the server is running. And before the configuration is loaded, we don't know if the server is running.

File: src/server.cpp [hunk 8]
```diff
@@ -68,20 +68,12 @@ void start_server(){
...
-        port = to_number<size_t>(config_value("server_port"));
-    }
-
-    std::string listen = "localhost";
-    if (config_contains("server_listen")) {
-        listen = config_value("server_listen");
-    }
-
+    auto port = get_server_port();
+    auto listen = get_server_listen();
```
Comment (wichtounet) [inline src/server.cpp]: Good :+1:

**Comment by wichtounet** [comment]
Yes, there are two different configuration files. Only .budgetrc should be touched by the user. The other configuration file is only to store internal data by budgetwarrior.

File: src/config.cpp (hunk 7 above)
Comment (robromano) [reply src/config.cpp]: Ah, that makes sense. Removed.

**Comment by wichtounet** [comment]
Thanks, I have merged your PR :)

**Pull request merged**